timeout when using multiple hosts in ES_HOSTS

## Describe the Bug
When specifying multiple hosts in ES_HOSTS, zipkin fails to connect to them. 

ES version is 7.2.0

openjdk version "9.0.4"
OpenJDK Runtime Environment (build 9.0.4+11)
OpenJDK 64-Bit Server VM (build 9.0.4+11, mixed mode)

## Steps to Reproduce
`STORAGE_TYPE=elasticsearch  ES_HOSTS=http://host1:9200,http://host2:9200 java -jar zipkin-server-2.16.1-exec.jar  
`

An exception is thrown about half a minute after the zipkin UI is accessed for the first time:

```
2019-08-20 14:39:26.844  INFO 14512 --- [           main] z.s.ZipkinServer                         : Started ZipkinServer in 3.986 seconds (JVM running for 5.296)
2019-08-20 14:39:48.260  WARN 14512 --- [cking-tasks-1-2] c.l.a.i.a.DefaultExceptionHandler        : [id: 0x15e9a196, L:/0:0:0:0:0:0:0:1:9411 - R:/0:0:0:0:0:0:0:1:54342][h1c://d02di1505646dit:9411/zipkin/api/v2/services#GET] Unhandled exception from an annotated service:

java.lang.IllegalStateException: couldn't connect any of []
        at zipkin2.server.internal.elasticsearch.LazyHttpClientImpl.decorateHealthCheck(LazyHttpClientImpl.java:134) ~[classes!/:?]
        at zipkin2.server.internal.elasticsearch.LazyHttpClientImpl.getEndpoint(LazyHttpClientImpl.java:95) ~[classes!/:?]
        at zipkin2.server.internal.elasticsearch.LazyHttpClientImpl.get(LazyHttpClientImpl.java:67) ~[classes!/:?]
        at zipkin2.elasticsearch.ElasticsearchStorage.http(ElasticsearchStorage.java:309) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.AutoValue_ElasticsearchStorage.http(AutoValue_ElasticsearchStorage.java:42) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.VersionSpecificTemplates.get(VersionSpecificTemplates.java:189) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.ElasticsearchStorage.ensureIndexTemplates(ElasticsearchStorage.java:279) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.AutoValue_ElasticsearchStorage.ensureIndexTemplates(AutoValue_ElasticsearchStorage.java:27) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.ElasticsearchStorage.spanStore(ElasticsearchStorage.java:209) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.elasticsearch.ElasticsearchStorage.serviceAndSpanNames(ElasticsearchStorage.java:214) ~[zipkin-storage-elasticsearch-2.16.1.jar!/:?]
        at zipkin2.server.internal.ZipkinQueryApiV2.getServiceNames(ZipkinQueryApiV2.java:83) ~[classes!/:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
        at com.linecorp.armeria.internal.annotation.AnnotatedHttpService.invoke(AnnotatedHttpService.java:260) ~[armeria-0.90.2.jar!/:?]
        at com.linecorp.armeria.internal.annotation.AnnotatedHttpService.lambda$serve0$3(AnnotatedHttpService.java:246) ~[armeria-0.90.2.jar!/:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642) ~[?:?]
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
        at com.linecorp.armeria.common.AbstractRequestContext.lambda$makeContextAware$1(AbstractRequestContext.java:69) ~[armeria-0.90.2.jar!/:?]
        at io.micrometer.core.instrument.AbstractTimer.record(AbstractTimer.java:160) ~[micrometer-core-1.2.0.jar!/:1.2.0]
        at io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:144) ~[micrometer-core-1.2.0.jar!/:1.2.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) [?:?]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.39.Final.jar!/:4.1.39.Final]
        at java.lang.Thread.run(Thread.java:844) [?:?]
Caused by: java.util.concurrent.TimeoutException
        at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886) ~[?:?]
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021) ~[?:?]
        at com.linecorp.armeria.client.endpoint.EndpointGroup.awaitInitialEndpoints(EndpointGroup.java:76) ~[armeria-0.90.2.jar!/:?]
        at zipkin2.server.internal.elasticsearch.LazyHttpClientImpl.decorateHealthCheck(LazyHttpClientImpl.java:131) ~[classes!/:?]
```
The classic UI will show this:
![image](https://user-images.githubusercontent.com/193792/63348376-637aab00-c359-11e9-9b7a-b1376ecc11f6.png)

Lens does not show (have ?) the error.

Maybe related during startup there is also this:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by io.netty.resolver.dns.DnsNameResolver (jar:file:/C:/tmp/zipkin-server-2.16.1-exec.jar!/BOOT-INF/lib/netty-resolver-dns-4.1.39.Final.jar!/) to method sun.net.dns.ResolverConfiguration.open()
WARNING: Please consider reporting this to the maintainers of io.netty.resolver.dns.DnsNameResolver
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

--- a/zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/InitialEndpointSupplier.java
+++ b/zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/InitialEndpointSupplier.java
@@ -104,12 +104,15 @@ final class InitialEndpointSupplier implements Supplier<EndpointGroup> {
     final CompletableFuture<List<Endpoint>> initialEndpointsFuture;
 
     volatile List<Endpoint> merged = Collections.emptyList();
-    volatile int dirty;
+    volatile int dirty = 1;
 
     CompositeEndpointGroup(List<EndpointGroup> endpointGroups) {
       this.endpointGroups = endpointGroups;
       for (EndpointGroup group : endpointGroups) {
-        group.addListener(unused -> notifyListeners(endpoints()));
+        group.addListener(unused -> {
+          dirtyUpdater.set(this, 1);
+          notifyListeners(endpoints());
+        });
       }
 
       initialEndpointsFuture = CompletableFuture.anyOf(

--- a/zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/InitialEndpointSupplierTest.java
+++ b/zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/InitialEndpointSupplierTest.java
@@ -1,0 +1,41 @@
+package zipkin2.server.internal.elasticsearch;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class InitialEndpointSupplierTest {
+  static final Endpoint FOO = Endpoint.of("foo");
+  static final Endpoint BAR = Endpoint.of("bar");
+  static final Endpoint CAT = Endpoint.of("cat");
+  static final Endpoint DOG = Endpoint.of("dog");
+
+  // Simple class to give public access to methods.
+  static class PublicDynamicEndpointGroup extends DynamicEndpointGroup {
+    void doSetEndpoints(Endpoint... endpoints) {
+      setEndpoints(Arrays.asList(endpoints));
+    }
+  }
+
+  @Test void compositeEndpoints() {
+    PublicDynamicEndpointGroup group1 = new PublicDynamicEndpointGroup();
+    PublicDynamicEndpointGroup group2 = new PublicDynamicEndpointGroup();
+
+    InitialEndpointSupplier.CompositeEndpointGroup composite =
+      new InitialEndpointSupplier.CompositeEndpointGroup(Arrays.asList(group1, group2));
+    assertThat(composite.endpoints()).isEmpty();
+
+    group1.doSetEndpoints(FOO, BAR);
+    assertThat(composite.endpoints()).containsExactlyInAnyOrder(FOO, BAR);
+    // Return memoized endpoints.
+    assertThat(composite.endpoints()).isSameAs(composite.endpoints());
+    group2.doSetEndpoints(CAT, DOG);
+    assertThat(composite.endpoints()).containsExactlyInAnyOrder(FOO, BAR, CAT, DOG);
+    group1.doSetEndpoints(FOO);
+    assertThat(composite.endpoints()).containsExactlyInAnyOrder(FOO, CAT, DOG);
+  }
+
+}

